extras/indexing/src/main/java/org/apache/rya/indexing/entity/EntityToSegmentConverter.java

```diff
@@ -0,0 +1,109 @@
+package org.apache.rya.indexing.entity;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.rya.indexing.entity.query.EntityQueryNode;
+import org.apache.rya.indexing.external.matching.ExternalSetConverter;
+import org.apache.rya.indexing.external.matching.JoinSegment;
+import org.apache.rya.indexing.external.matching.OptionalJoinSegment;
+import org.apache.rya.indexing.external.matching.QuerySegment;
+import org.openrdf.query.algebra.Filter;
+import org.openrdf.query.algebra.Join;
+import org.openrdf.query.algebra.LeftJoin;
+import org.openrdf.query.algebra.QueryModelNode;
+import org.openrdf.query.algebra.TupleExpr;
+import org.openrdf.query.algebra.ValueExpr;
+import org.openrdf.query.algebra.helpers.QueryModelVisitorBase;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * Implementation of {@link ExternalSetConverter} to convert {@link EntityQueryNode}s
+ * to {@link QuerySegment}s.
+ *
+ */
+public class EntityToSegmentConverter implements ExternalSetConverter<EntityQueryNode> {
+
+    private static final EntityToOptionalJoinSegment optional = new EntityToOptionalJoinSegment();
+    private static final EntityToJoinSegment join = new EntityToJoinSegment();
+
+
+    @Override
+    public QuerySegment<EntityQueryNode> setToSegment(final EntityQueryNode set) {
+        Preconditions.checkNotNull(set);
+        //if (PCJOptimizerUtilities.tupleContainsLeftJoins(set.getTupleExpr())) {
+            return optional.getSegment(set);
+       // } else {
+          //  return join.getSegment(set);
+        //}
+    }
+
+    /**
+     * This class extracts the {@link JoinSegment} from the {@link TupleExpr} of
+     * specified PCJ.
+     *
+     */
+    static class EntityToJoinSegment extends QueryModelVisitorBase<RuntimeException> {
+
```

Comment:
This may be an artifact of what I did before and after your refactor.  if its isolated not used code maven is happy with it.  same below